src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts

```diff
@@ -8,101 +8,374 @@ import {
-import { IronAutogrowTextareaElement } from "@polymer/iron-autogrow-textarea";
-import "@polymer/paper-input/paper-textarea";
 import "../../../../../components/ha-circular-progress";
-import { LOG_OUTPUT, reconfigureNode } from "../../../../../data/zha";
+import "../../../../../components/ha-svg-icon";
+import {
+  AttributeConfigurationStatus,
+  Cluster,
+  ClusterConfigurationEvent,
+  ClusterConfigurationStatus,
+  fetchClustersForZhaNode,
+  reconfigureNode,
+  ZHA_CHANNEL_CFG_DONE,
+  ZHA_CHANNEL_MSG_BIND,
+  ZHA_CHANNEL_MSG_CFG_RPT,
+} from "../../../../../data/zha";
 import { fireEvent } from "../../../../../common/dom/fire_event";
+import { UnsubscribeFunc } from "home-assistant-js-websocket";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
 
 @customElement("dialog-zha-reconfigure-device")
 class DialogZHAReconfigureDevice extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @internalProperty() private _active = false;
+  @internalProperty() private _status?: string;
+
+  @internalProperty() private _stages?: string[];
+
+  @internalProperty() private _clusterConfigurationStatuses?: Map<
+    number,
+    ClusterConfigurationStatus
+  > = new Map();
 
-  @internalProperty() private _formattedEvents = "";
+  @internalProperty() private _params:
+    | ZHAReconfigureDeviceDialogParams
+    | undefined = undefined;
 
-  @internalProperty()
-  private _params: ZHAReconfigureDeviceDialogParams | undefined = undefined;
+  @internalProperty() private _allSuccessful = true;
 
-  private _subscribed?: Promise<() => Promise<void>>;
+  @internalProperty() private _showDetails = false;
 
-  private _reconfigureDeviceTimeoutHandle: any = undefined;
+  private _subscribed?: Promise<UnsubscribeFunc>;
 
-  public async showDialog(
-    params: ZHAReconfigureDeviceDialogParams
-  ): Promise<void> {
+  public showDialog(params: ZHAReconfigureDeviceDialogParams): void {
     this._params = params;
-    this._subscribe(params);
+    this._stages = undefined;
   }
 
   public closeDialog(): void {
     this._unsubscribe();
-    this._formattedEvents = "";
     this._params = undefined;
+    this._status = undefined;
+    this._stages = undefined;
+    this._clusterConfigurationStatuses = undefined;
     fireEvent(this, "dialog-closed", { dialog: this.localName });
   }
 
   protected render(): TemplateResult {
     if (!this._params) {
       return html``;
     }
+
     return html`
       <ha-dialog
         open
-        hideActions
-        @closing="${this.closeDialog}"
+        @closed="${this.closeDialog}"
         .heading=${createCloseHeading(
           this.hass,
-          this.hass.localize(`ui.dialogs.zha_reconfigure_device.heading`)
+          this.hass.localize(`ui.dialogs.zha_reconfigure_device.heading`) +
+            ": " +
+            (this._params?.device.user_given_name || this._params?.device.name)
```

Comment:
`${this.hass.localize(`ui.dialogs.zha_reconfigure_device.heading`)}: ${(this._params?.device.user_given_name || this._params?.device.name)}`